### DISCUSSION TITLE
YAMCS and openMCT plugin fail to integrate, how do I resolve?

### DISCUSSION THREAD

**Discussion by JohnAdam-123** [nasa/openmct#7920] [Community Q&A] [answered]
I have been able to install yamcs quick-start project and ran it in my computer, also I managed to install openMCTC-YAMCS and ran it. But the problem is that openmct web interface has not been able to connect with the YAMCS server running. How is that solved?

**Comment by thstasy** [comment]
Could you elaborate your issue? I have the same. But mine is that, on the web interface of OpenMCT, there is this "i.valueForHints() is not a function"

**Comment by jvigliotta** [comment]
Hello! Sorry for the delay in response. I'm going through the steps to see if I encounter this issue as well, I'll log my actions below:

Starting here: https://github.com/akhenry/openmct-yamcs
```
git clone https://github.com/akhenry/openmct-yamcs.git
cd openmct-yamcs
npm install
npm run build:example
npm start
```

Visit: http://localhost:9000/

I'm seeing errors in the terminal, this is because the pre-requisite https://github.com/yamcs/quickstart hasn't been set up.
I clone the repo: `git clone https://github.com/yamcs/quickstart.git` in a different folder from my openmct-yamcs repo.
Enter the quickstart folder: `cd quickstart`
Run the commands from the quickstart readme (linked above).
```
./mvnw compile
./mvnw yamcs:run
```

In a new terminal, back in the quickstart folder I was just in, I run:
`python simulator.py`
This fails for me, my python is `python3`
I run `python3 simulator.py`
It shows packets are flowing now.

Refresh: `http://localhost:9000/`

Now I'm seeing telemetry when navigating the object tree on the left.

Try these steps and let us know if you run into any errors.

**Comment by thstasy** [comment]
Hi Jamie,

Thanks for the help!

I have a few more follow-up questions, and would greatly appreciate any
further insights.

Subject: Questions on UDP Message Handling, Web Interface Customization,
and Data Archiving in Yamcs.


*UDP Message Parameters*

I’m having the following error when sending a UDP message:

WARNING: Exception when receiving parameter data:
com.google.protobuf.InvalidProtocolBufferException: Cannot find field:
description in message yamcs.protobuf.pvalue.ParameterValue.

I went through the .proto files but not much relevant information found.
Could you clarify what parameters are allowed in a UDP message, and where I
might find the correct field definitions?


*Adding a "Timestamp" Column to the Parameters Web Interface*

I want to add a "Timestamp" column (alongside the existing columns: "Name",
"Value", "Data Source", "Description") on the Parameters web interface.
Where should I start to implement this?


*Data Archiving Issue*

I have updated the parameters' values, and the changes are reflected in the
"Value" column on the web interface and in the curl command output.
However, when I query the archive with curl -s "
http://localhost:8090/api/archive/UEI_DAQ/parameters/UEI_DAQ/AI_ch0?limit=10
", it shows data from 3 days ago instead of the latest value. Could you
help me troubleshoot why the archive is not reflecting the most recent
updates?


Thank you for your time and assistance!


Best regards,

Stasy

Jamie V. ***@***.***> 於 2025年4月2日 週三 02:37 寫道：

**Comment by jvigliotta** [comment, marked as answer]
Hi @thstasy 

These issues are outside of my realm of knowledge, they seem more YAMCS based, you may be able to find answers at the YAMCS repos: https://github.com/yamcs

Sorry I'm not able to provider more information on those issues.
